test/builder_quantization.cpp

```diff
@@ -1182,6 +1184,31 @@ TEST(builder, scaled_quantize_concat_unsigned_varying)
               read_vector<uint8_t>(result));
 }
 
+TEST(builder, scaled_QDotInteger)
+{
+    Shape shape_a{3, 3}; // input shape
```

Comment:
If you named your variable `input_shape` then the comment would not be needed and the code would be that much more readable. You could also rename a_data to input_data, etc..

Reply:
It would be `input/data_shape`, `filter_shape`, `output/result_shape` and `input/data`, `filter`, `output/result`, but not `input_data` which is redundant.